codalab/worker_manager/worker_manager.py

```diff
@@ -147,25 +147,24 @@ def build_command(self, worker_id: str, work_dir: str) -> List[str]:
         return command
 
     def run_loop(self):
-        while True:
-            try:
-                self.run_one_iteration()
-            except (
-                urllib.error.URLError,
-                http.client.HTTPException,
-                socket.error,
-                NotFoundError,
-                JsonApiException,
-            ):
-                # Sometimes, network errors occur when running the WorkerManager . These are often
-                # transient exceptions, and retrying the command would lead to success---as a result,
-                # we ignore these network-based exceptions (rather than fatally exiting from the
-                # WorkerManager )
-                traceback.print_exc()
-            if self.args.once:
-                break
-            logger.debug('Sleeping {} seconds'.format(self.args.sleep_time))
-            time.sleep(self.args.sleep_time)
+        try:
```

Comment:
If you get rid of the `while True` loop here, it will only one iteration. Is this intended? That doesn't seem right.

Reply:
That one iteration could still produce errors and we need to handle them. What's the problem?

Reply:
The worker manager loops infinitely checking if it should spawn a worker. By getting rid of the `while True` loop it will only do this once.

Reply:
Just to be more clear, logging into CodaLab is not the only functionality of `run_one_iteration()`. I would take a look at what the method does.

Reply:
Gotcha, I put back the loop and added some additional changes

Reply:
I think this came from a discussion where if login fails, then you just exit without trying indefinitely, so it's easier to test the code.

Reply:
Yes I remember it. So the most recent code only allows login attempt once and maintain other functionalities of `run_one_iteration()`